controller/src/main/java/org/jboss/as/controller/CapabilityReferenceRecorder.java

```diff
@@ -47,6 +47,10 @@
      */
     void removeCapabilityRequirements(OperationContext context, String attributeName, String... attributeValues);
 
+    String getBaseDependentName();
+
+    String getBaseRequirementName();
+
```

Comment:
I will check with elytron guys to see how their custom impl looks like and if we can abolish the CapabilityReferenceRecorder interface and hide all its stuff internally